### DISCUSSION TITLE
use-places-autocomplete hook in react hook form

### DISCUSSION THREAD

**Discussion by sunilbittu** [react-hook-form/react-hook-form#2533] [Q&A] [answered]
Hi All, anyone tried using use-places-autocomplete in react-hook-form with useFieldarray.

https://github.com/wellyshen/use-places-autocomplete 

Is it possible to implement?

Tried like this but all input fields showing same value
https://codesandbox.io/s/react-hook-form-usefieldarray-9qvlb?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
I think you should build a separate component and wrap it with Controller.

**Reply by sunilbittu** [reply]
Ok bill I'll try

**Reply by sunilbittu** [reply]
hi @bluebill1049 i have tried using the controller but address is coming as undefined.
![image](https://user-images.githubusercontent.com/10049424/89755955-c29f7580-dafe-11ea-98d4-77fa762dbd44.png)

https://codesandbox.io/s/react-hook-form-usefieldarray-9qvlb?file=/src/search.js
is there anyway to get address value

**Reply by bluebill1049** [reply]
<img width="1313" alt="Screen Shot 2020-08-10 at 4 15 51 pm" src="https://user-images.githubusercontent.com/10513364/89756066-c7772000-db24-11ea-91e6-29a8c1443dde.png">

hey @sunilbittu you forgot to save the CSB?

**Reply by sunilbittu** [reply]
my bad ! 
https://codesandbox.io/s/react-hook-form-usefieldarray-9qvlb?file=/src/search.js

**Reply by bluebill1049** [reply]
no worries, have a read at the doc on `as` vs `render`, I think it's important to under the mechanism behind https://react-hook-form.com/api#Controller

`as` will pass `onChange` `onBlur`, `value` props, you need to wire them up.

`render` is a render prop which you can wire up inside the callback function.

https://codesandbox.io/s/react-hook-form-usefieldarray-7bpmk?file=/src/search.js

once you understand the above, it should be pretty easy to integrate with any components.
